pkg/kf/commands/utils/utils.go

```diff
@@ -122,3 +125,39 @@ func parseCommandLine() (args []string, flags map[string][]string) {
 	}
 	return args, flags
 }
+
+// CreateProxy creates a proxy to the specified gateway with the specified host in the request header.
+func CreateProxy(w io.Writer, host, gateway string) *httputil.ReverseProxy {
+	logger := log.New(w, fmt.Sprintf("\033[34m[%s via %s]\033[0m ", host, gateway), log.Ltime)
+
+	return &httputil.ReverseProxy{
+		Director: func(req *http.Request) {
+			req.Host = host
+			req.URL.Scheme = "http"
+			req.URL.Host = gateway
+
+			logger.Printf("%s %s\n", req.Method, req.URL.RequestURI())
+		},
+		ErrorLog: logger,
+	}
+}
+
+// PrintCurlExamples lists example HTTP requests the user can send.
+func PrintCurlExamples(w io.Writer, listener net.Listener, routeHost, gateway string, withProxy bool) {
```

Comment:
This is a little weird to be an exported function.

Reply:
this and the CreateProxy method are being used in both apps and routes now, I'll move this over to the internal utils package in a separate PR